helpers/mathjax.js

```diff
@@ -186,6 +191,17 @@ export function loadMathJax(mathJaxConfig) {
 					const InputJax = startup.getInputJax();
 					const OutputJax = startup.getOutputJax();
 					const html = mathjax.document(root, { InputJax, OutputJax });
+
```

Comment:
Any chance you can remind me what the difference is between these two calls?

Reply:
Yeah, so the second function is one I had to newly add to capture the input/output Jax settings properly. MathJax's `typesetPromise` API seems to use their default document for reasons. For non-deferred rendering cases we're not actually in a shadowRoot (or we can't really determine if we are), so instead we just pass in the elements we care about rendering and take the document itself as the parent.

Reply:
A perhaps more elegant solution may be to check if the element is in a shadowRoot, and if it is, pass that shadowRoot into the first element, add the `elem` to typeset as an optional argument, and then just combine it into one call.

Reply:
Just to clarify, in what cases would we not be able to determine if we're in a shadow DOM?  Assuming the element is connected, `node.getRootNode()` could always be compared to the `document` element.  If it's different, it's a shadow DOM.

I am mildly concerned about the case where an `d2l-html-block` is inside the shadowDOM of an app component.

When we use `html.options.elements = [elem];`, is that effectively telling MathJax... typeset the elements in this array?

Reply:
Yes, the `no-deferred-rendering` case has never worked inside the shadowDOM of an app component. I believe we didn't care when first implementing this because we didn't actually want anybody to use it. It's not particularly difficult to make it work, it's just something we explicitly didn't care for supporting before.

Yes, `html.options.elements = [elem];` limits which elements we want MathJax to typeset.

Reply:
Thanks, I remember that conversation now.  😄

Reply:
Yeah, given that we can't use MathJax's API anyways, I did just push a commit that added this support. It's almost more work not to do it at this point, haha.